modules/javafx.graphics/src/main/java/com/sun/javafx/geom/Dimension2Df.java

```diff
@@ -26,18 +26,9 @@
 package com.sun.javafx.geom;
 
 /**
- * The <code>Dimension2D</code> class is to encapsulate a width
- * and a height dimension.
- * <p>
+ * A 2D dimension object that contains a width and a height as floats.
+ *
+ * @see com.sun.javafx.geom.Rectangle.Dimension2Di Dimension2Di - ints
+ * @see javafx.geometry.Dimension2D Dimension2D - doubles
  */
-public class Dimension2D {
-    public float width;
-    public float height;
-
-    public Dimension2D() { }
-
-    public Dimension2D(float w, float h) {
-        width = w;
-        height = h;
-    }
-}
+public record Dimension2Df(float width, float height) {}
```

Comment:
is this class really needed?

Looks like it's only referenced from `Arc2D::setArc()` and `RectanglularShape::setFrame()`, as far as I can tell these two methods are never called.

Should all three be removed?

Reply:
I'm not familiar with these methods, but I did wonder why we need extra implementations in the initial comment.

Reply:
looks like dead code left over from the original implementation.

I removed the class along with the two methods and jfx builds and runs (i.e. shows arcs and rectangles) just fine.  Eclipse IDE also has no issues.  I think we can remove these.

Reply:
I don't mind removing them in this PR if others agree.